Oracle.md

```diff
@@ -0,0 +1,190 @@
+
+### Serialization of event descriptors
+
+Event descriptors should be serialized using [TLV format](https://github.com/lightningnetwork/lightning-rfc/blob/master/01-messaging.md#type-length-value-format) as described bellow.
+
+#### Version 0 `enum_event_descriptor`
+
+1. type: 55302 (`enum_event_descriptor_v0`)
+2. data:
+   * [`x_point`:`oracle_nonce`]
+   * [`u16`:`num_outcomes`]
+   * [`u16`:`outcome_1_len`]
+   * [`string`:`outcome_1`]
+   * ...
+   * [`u16`:`outcome_n_len`]
+   * [`string`:`outcome_n`]
+
+This type of event descriptor is a simple enumeration where the value `n` is the number of outcomes in the event.
+
+Note that `outcome_i` is the outcome value itself and not its hash that will be signed by the oracle.
+
+#### Version 0 `range_event_descriptor`
+
+1. type: 55304 (`range_event_descriptor_v0`)
+2. data:
+   * [`x_point`:`oracle_nonce`]
+   * [`int32`:`start`]
+   * [`u32`:`count`]
+   * [`u16`:`step`]
+   * [`string`:`unit`]
+   * [`int32`:`precision`]
+
+#### Version 0 `large_range_event_descriptor`
+
+1. type: 55306 (`large_range_event_descriptor_v0`)
+2. data:
+   * [`bigsize`:`base`]
+   * [`bool`:`is_signed`]
+   * [`u16`:`nb_nonces`]
+   * [`nb_nonces*x_point`:`oracle_nonces`]
+   * [`string`:`unit`]
+   * [`int32`:`precision`]
+
+### Oracle events
+
+For users to be able to create DLCs based on a given event, they also need to obtain information about the oracle and the time at which it plans on releasing a signature over the event outcome.
+Oracle events contain such information, which includes:
+* the oracle public key,
+* the earliest time (UTC) at which it plans on releasing a signature over the event outcome, in epoch seconds,
+* the event descriptor,
+* the event URI which can be a name or categorization associated with the event by the oracle.
+
+The TLV serialization for oracle events is as follow:
+
+#### Version 0 `oracle_event`
+
+1. type: 55330 (`oracle_event_v0`)
+2. data:
+   * [`x_point`:`oracle_public_key`]
+   * [`u32`:`event_maturity_epoch`]
+   * [`event_descriptor`:`event_descriptor`]
+   * [`string`:`event_uri`]
+
+## Oracle announcements
+
+In order to make it possible to hold oracles accountable in cases where they do not release a signature for an event outcome, there needs to be a proof that an oracle has committed to a given outcome.
+This proof is given in a so-called oracle announcement, which contains an oracle event together with a signature over its serialization, created using the same private key that the oracle plans on signing the event outcome with.
+
+This also makes it possible for users to obtain oracle event information from an un-trusted peer while being guaranteed that it originates from a given oracle.
+
+The TLV serialization of oracle announcements is as follow.
+
+#### Version 0 `oracle_announcement`
+
+1. type: 55332 (`oracle_announcement`)
+2. data:
+   * [`signature`:`annoucement_signature`]
+   * [`oracle_event`:`oracle_event`]
```

Comment:
What are the requirements of a receiver of such `oracle_announcement`, independently of the way it has obtained. I think it should precise that oracle signature must be verified against oracle pubkey (a subset of #97 scope)

How do you learn about an oracle pubkey, without doxxing yourself is another concern we can address later.

Reply:
Don't you think the sentence:
>This proof is given in a so-called oracle announcement, which contains an oracle event together with a signature over its serialization, created using the same private key that the oracle plans on signing the event outcome with.

Already imply that the signature must verify against the oracle public key? I can add more to it but it seems already pretty clear to me. Maybe what and when to verify should be more in the protocol spec I think.

Reply:
Yes I think that there is no need to add more here, though I agree that it wouldn't hurt for @ariard to include mention of this verification along with all the other ones when he writes the client validation spec

Reply:
Okay I'll take it in #97